### PR TITLE
Add feature flag for report builder beta group

### PR DESCRIPTION
**Pull request by NoahCarnahan**
The report builder beta group needs a feature flag separate from the standard report builder feature flag because domains in the beta group should be able to access the report builder even if they aren't on a subscription that grants report builder privileges.

cc @nickpell

### CODE REVIEW

File: corehq/apps/userreports/views.py [hunk 1]
```diff
@@ -144,10 +146,14 @@ class ReportBuilderView(BaseDomainView):
     @use_select2
     @use_daterangepicker
     @use_datatables
-    @method_decorator(toggles.REPORT_BUILDER.required_decorator())
-    @method_decorator(requires_privilege_raise404(privileges.REPORT_BUILDER))
```
Comment (NoahCarnahan) [inline corehq/apps/userreports/views.py]: The conditional logic requires moving these checks out of the decorators and into the function itself.

File: corehq/apps/userreports/views.py [hunk 2]
```diff
@@ -446,7 +452,7 @@ def _edit_report_shared(request, domain, config, read_only=False):
     return render(request, "userreports/edit_report_config.html", context)
 
 
-@toggles.any_toggle_enabled(toggles.USER_CONFIGURABLE_REPORTS, toggles.REPORT_BUILDER)
+@toggles.any_toggle_enabled(toggles.USER_CONFIGURABLE_REPORTS, toggles.REPORT_BUILDER, toggles.REPORT_BUILDER_BETA_GROUP)
```
Comment (dimagimon) [inline corehq/apps/userreports/views.py]: E501 line too long (121 > 115 characters)

File: corehq/apps/hqwebapp/models.py [hunk 3]
```diff
@@ -295,8 +295,11 @@ def sidebar_items(self):
         ])]
 
         user_reports = []
-        if (toggle_enabled(self._request, toggles.REPORT_BUILDER)
-                and has_privilege(self._request, privileges.REPORT_BUILDER)):
+
+        builder_enabled = toggle_enabled(self._request, toggles.REPORT_BUILDER)
+        builder_privileges = has_privilege(self._request, privileges.REPORT_BUILDER)
+        beta_group_enabled = toggle_enabled(self._request, toggles.REPORT_BUILDER_BETA_GROUP)
+        if ((builder_enabled and builder_privileges) or beta_group_enabled):
```
Comment (nickpell) [inline corehq/apps/hqwebapp/models.py]: would be nice if the logic here was moved to its own function to reduce code duplication

Comment (NoahCarnahan) [inline corehq/apps/hqwebapp/models.py]: will do

**Comment by NoahCarnahan** [comment]
@nickpell updated

**Pull request merged**